### ISSUE TITLE
WP - Performance issue (found something)

### ISSUE THREAD

**Issue by graphefruit** [angular/material#8218]
Hello Angular-Team,

I'm having big troubles with the performance on Windows Phones (WP 8.1 aswell as on WP 10).
But I've found a big "problem", and got a poor but working solution to this.

The initial steps are described here from Thomas: https://github.com/angular/material/issues/1771#issuecomment-101070668

I overwrote `$MD_THEME_CSS`
`$provide.constant('$MD_THEME_CSS', '');`

After this changes, my App looked terrible, but the performance was drastically improved. 
**Example:** My newspage had a load time from **10 to 11** seconds, after this change the page loaded **4 seconds**.

The problem is, that IE having problems with the large inline-stylesheets which are rendered into the DOM.
![image](https://cloud.githubusercontent.com/assets/6260244/14814575/2b17fb86-0ba7-11e6-8bca-1c473166157f.png)

So my solution was to copy and paste all of the rendered styles, and put them into a single `.CSS`
And reference this in my `.HTML`

After this, my App looked like before (didn't notice any problems yet) and the load time is still 4 seconds.

@ThomasBurleson could you please take a look? 
Maybe a complete `.CSS` could be shipped? 

@topherfangio #7659 

Thanks in advance for your time.
Greetings
Graphefruit

**Comment by ThomasBurleson** [comment]
@EladBezalel - perhaps we should have a $mdThemingProvider configuration option to use only 1 single stylesheet injection; default == false (since some browsers want smaller stylesheets).

**Comment by graphefruit** [comment]
First: Thanks for feedback!
Haven't testet it actually with one stylesheet only.
I gonna take a look and reporting if this will help or if still the inline-styles would be a problem.

Greetings

**Comment by graphefruit** [comment]
Tested it now a little bit (10 minutes)

The performance was increased aswell when the styles were bunched in on style tag.
But the performance was bigger, in my opinion, when I excluded the styles to one single `.CSS-File`

I also tested on iOS and Android, and those got also a performance improvements from this.

So maybe there could be 3 ways?
- Multiple styles
- One big style
- All styles in a .CSS (if possible?)

I gonna test tomorrow a little bit more.
I had a briefe lag while having all in one big style tag, but I can't exclude if this was maybe a problem of my app.

Thanks so far
Graphefruit

**Comment by EladBezalel** [comment]
@graphefruit i don't think a `css file` is possible because it's being built on the fly.

A one big style would be the best solution as i can figure it right now,

@ThomasBurleson another thing that can fit here is selective builds,
allowing users to specify the components they're using and building only the css related

**Comment by graphefruit** [comment]
@EladBezalel I understand, thanks so far. Maybe there could be three options:
1. Generate no style->I would use this one and insert this in my own `.CSS` extracted from number 2.
2. Generate 1 style
3. Generate multiple styles

+1 for selective builds.

**Comment by EladBezalel** [comment]
@graphefruit i don't think we should support `Generate no style` as we are a ux framework and most of our components relay on these classes..

**Comment by graphefruit** [comment]
@EladBezalel so atleast don't remove `$provide.constant('$MD_THEME_CSS', '');` and I would be more then happy ;)

Just give me a call if I can test something for you on my real WP devices.

**Comment by EladBezalel** [comment]
`$provide.constant('$MD_THEME_CSS', '');` is a valid workaround, we won't restrict it.

**Comment by david-gang** [comment]
Hi @EladBezalel , I agree that by default the css style should be injected.

Additionally maybe a gulp task could be provided which basically runs the logic of the css style injection and generate a css file. This, togethere with setting $provide.constant('$MD_THEME_CSS', '');  could be a valid workaround.

What do you think regarding this solution?

Thanks,
David

**Comment by EladBezalel** [comment]
By providing a gulp task we force the user who want this to be dependent on gulp. 
As I said earlier, supporting it out of the box doesn't sounds appropriate for a UX framework, 
Running it by yourself self for the first time and copying that is the valid workaround

**Comment by david-gang** [comment]
@EladBezalel You are correct.

I wrote an one liner which copies all the stylesheet data to the clipboard. This works in chrome developer tools
```javascript
copy([].slice.call(document.styleSheets).map(e=>e.ownerNode).filter(e => e.attributes && e.attributes[0].nodeName==='md-theme-style').map(e => e.textContent).join("\n"))
```

**Comment by Splaktar** [comment]
@graphefruit another option in 1.1.0 is to use `$mdThemingProvider.disableTheming();` instead of $provide.constant('$MD_THEME_CSS', '');.

**Comment by graphefruit** [comment]
Why is this deprecated now? 
May you need some more information?

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.
